### DISCUSSION TITLE
Have this work on video and audio files as well

### DISCUSSION THREAD

**Discussion by SaadBazaz** [cedricbonhomme/Stegano#39] [Ideas]
Hello,
Image steganography is cool and all... what if this supported video and audio directly, as well?

**Comment by cedricbonhomme** [comment]
Hi,
Maybe you know a technique or algorithm for audio and/or video files?
Apart from using the metadata fields from the files (not really interesting) I have no idea how to do this concretely.

**Reply by shashanknimje** [reply]
Thanks much for your work. I used ffmpeg to extract frames from a short video clip and watermarked each frame using stegano. I can verify that the individual frames do have the watermark data. 

But when I use ffmpeg to re-render a video using each of those watermarked frames, and then extract a random frame from the output video, the extracted frame does not have the embedded watermark anymore. 

Would you happen to know why that might be happening? I do check the video codec, resolution, fps, video bitrate, audio codec and bitrate and use those parameters at the time of re-rending the output video using the individual watermarked frames. Any help would be much appreciated.

**Comment by cedricbonhomme** [comment]
The release 2.0.0 of Stegano introduces a way to hide/reveal messages in PCM encoded .wav files.
